### ISSUE TITLE
aws synthetics: cleanup canary resources

### ISSUE THREAD

**Issue by lukasnagl** [aws/aws-cdk#29548]
### Describe the feature

Upon stack deletion I noticed that `Canary` constructs left behind not only lambda functions, but S3 buckets (containing screenshots) as well. I would have expected all resources created by the canary to be deleted automatically, or at least having a `retention` option on the `Canary` construct. 

### Use Case

On deletion of a `Canary` construct, all resources (lambdas, s3 buckets,…) created for it should be deleted by default.

### Proposed Solution

I believe #26580 went for a solution approach, but is limited to the lambda function for the canary. Maybe this can be extended to all resources, or even made the default instead of a separate `cleanup` configuration?

### Other Information

I saw similar efforts in #18448, which added lambda deletion but did not resolve s3 bucket deletion to my knowledge.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.130.0

### Environment details (OS name and version, etc.)

irrelevant

**Comment by pahud** [comment]
Yes we probably can do that from here

https://github.com/aws/aws-cdk/blob/0fee99b473c17210a156d74263227cdf3274d253/packages/aws-cdk-lib/aws-synthetics/lib/canary.ts#L318-L320

**Comment by SigniantJoe** [comment]
I was just about to create the same feature request, but in my mind the real issue is Log Groups rather than S3 buckets.

Here's my thoughts on the matter with a TLDR at the end:

### S3 Buckets
From what I've seen, by default Canaries of a region all use one shared bucket by default(`cw-syn-results-{account_id}-{region}`). If you create the S3 Bucket separately then set the Canary `artifactsBucketLocation` parameter, that bucket is created as part of the stack and cleanup gets handled by CloudFormation.

If you're expecting CDK to clean up the default bucket for you, that's a bit complicated because other Canary tests could still be using it. A nice feature would be to pass Canary some logging properties and have it create a bucket on our behalf so that we don't have to separately create the execution role, but that's possibly out of scope.

### IAM execution role and policies
Canary will create one as part of the CloudFormation stack, which means it gets cleaned up.
Alternatively, the same as S3 Buckets, you can separately create the role with permissions to write to your bucket and set the `executionRoleArn` parameter.

### Log Groups
Unlike S3 buckets and IAM roles, there's no workaround to manually create the log group. CDK can't do it because AWS doesn't allow it and the name isn't deterministic and AWS. The only way to handle this currently would be to add log group removal to that cleanup function.

Related issue: https://github.com/aws/aws-cdk/issues/23718

### Alarms
CDK doesn't currently let you create Canary alarms, but the metric dimensions are deterministic using `CanaryName` so it's possible to create these separately and have CloudFormation clean them up.

### TLDR
- Canary cleanup should include log groups
- It would be nice if Canary cleanup includes S3 buckets, but the exact implementation would need further consideration and effort/benefit analysis
- Other resources are already handled appropriately.
